src/runtime_src/core/pcie/windows/core_system.cpp

```diff
@@ -92,15 +66,22 @@ static std::string getmachinename()
 void xrt_core::system::get_os_info(boost::property_tree::ptree &_pt)
 {
   char tnow[26];
-  time_t result = 0;
+  time_t result = time(NULL);
+  char value[BUFFER];
+  DWORD BufferSize = BUFFER;
 
   ctime_s(tnow, sizeof tnow, &result);
-  _pt.put("sysname", GetStringFromReg(HKEY_LOCAL_MACHINE,
-				"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", "ProductName").c_str());
-  _pt.put("release", GetStringFromReg(HKEY_LOCAL_MACHINE,
-				"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", "BuildLab").c_str());
-  _pt.put("version", GetStringFromReg(HKEY_LOCAL_MACHINE,
-				"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", "CurrentVersion").c_str());
+
+  RegGetValueA(HKEY_LOCAL_MACHINE, "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", "ProductName", RRF_RT_ANY, NULL, (PVOID)&value, &BufferSize);
```

Comment:
In the previous version of this submission, you had encapsulated the calls with a helper function.  You might wish to consider doing the same here and add the following DRCs:
1) Validate there is indeed a return value (e.g., the BufferSize is NOT zero ;)
2) Report a default value or no entry at all if the call failed.